xmodule/vertical_block.py

```diff
@@ -116,16 +129,16 @@ def _student_or_public_view(self, context, view):  # lint-amnesty, pylint: disab
                 child_block_context['wrap_xblock_data'] = {
                     'mark-completed-on-view-after-delay': complete_on_view_delay
                 }
-
-            try:
-                # .. filter_implemented_name: VerticalBlockChildRenderStarted
-                # .. filter_type: org.openedx.learning.vertical_block_child.render.started.v1
-                child, child_block_context = VerticalBlockChildRenderStarted.run_filter(
-                    block=child, context=child_block_context
-                )
-            except VerticalBlockChildRenderStarted.PreventChildBlockRender as exc:
-                log.info("Skipping %s from vertical block. Reason: %s", child, exc.message)
-                continue
+            if XBLOCK_SKILL_TAG_VERIFICATION_SWITCH.is_enabled():
```

Comment:
@irfanuddinahmad: Why was the entire running of the `VerticalBlockChildRenderStarted ` and `VerticalBlockRenderCompleted` filters now gated behind `XBLOCK_SKILL_TAG_VERIFICATION_SWITCH`? The filters are supposed to be a general extension mechanism for people to override unit rendering behavior, and is not specific to one feature.

@tecoholic: As the original author that [added the filters](https://github.com/openedx/edx-platform/pull/31388), please let me know if I'm reading this wrong.

FYI @nsprenkle, since this is an extension point you may want to use.

Reply:
Also FYI to @mariajgrimaldi ^

Reply:
@ormsbee Yes. `VerticalBlockChildRenderStarted` and `VerticalBlockRenderCompleted` are in fact meant to be generic filters. For eg., we use it in a course to automatically inject content like hyper-links to edit OLX source material stored in a Git repo like Gitlab/Github.

Reply:
@irfanuddinahmad: Also, when you're making a change like this, please put the reason why this change was done in the commit message. I'm assuming this was in response to some bug that was found, but I can't access that JIRA ticket to find out what it is or why it broke.

Reply:
@ormsbee ENT-7614 was in response to a downtime in Discovery and LMS due to delayed response originating from the XBlockSkillsAPI list endpoint that is responsible for gathering any skills linked to an xblock. This switch was added to quickly turn the feature on/off.

Reply:
@irfanuddinahmad: Thank you for the quick response. I think I understand what happened now.

**This needs to get reverted.** If 2U needs to be able to toggle off a plugin to make things more operationally robust, then the toggle and the conditional code should go inside that plugin. We absolutely cannot have this switch disable the entire filter pipelines that the plugin uses, since that will break everyone else who uses these extension points. The fact that this regression is hidden because the tests for it were altered to run with that switch enabled is also not an acceptable solution, since most people won't be running their instances that way.

This change made it into the Quince release. Per the [release schedule](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3613392957/Open+edX+Release+Schedule), the next patch to Quince will be released on 2024-04-09.

**Please have these changes reverted in master and back-ported into quince.master by 2024-04-02.**

I want to give you folks the time to make a smooth transition here, but this is a breaking bug in Quince, and we need to get it patched.

Please reach out to me if you have any questions or concerns.

FYI @georgebabey: I'm making a request of your team here ⬆️. 

Thanks folks.

Reply:
@ormsbee Here are the two PRs
https://github.com/openedx/edx-platform/pull/34348
https://github.com/openedx/edx-platform/pull/34347